pkgs/development/libraries/ffmpeg/0001-avfoundation.m-macOS-SDK-10.12-compatibility.patch

```diff
@@ -0,0 +1,25 @@
+From a804dd05351716b1fd9a7c2b7ed636613edd9c0f Mon Sep 17 00:00:00 2001
+From: toonn <toonn@toonn.io>
+Date: Fri, 21 Jun 2024 18:59:49 +0000
+Subject: [PATCH] avfoundation.m: macOS SDK 10.12 compatibility
+
+---
+ libavdevice/avfoundation.m | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/libavdevice/avfoundation.m b/libavdevice/avfoundation.m
+index d9b17ccdae..2a1f66d751 100644
+--- a/libavdevice/avfoundation.m
++++ b/libavdevice/avfoundation.m
+@@ -762,7 +762,7 @@ static int get_audio_config(AVFormatContext *s)
+     return 0;
+ }
+ 
+-static NSArray* getDevicesWithMediaType(AVMediaType mediaType) {
++static NSArray* getDevicesWithMediaType(NSString * mediaType) {
```

Comment:
For how long will this be forward compatible?

Reply:
It looks like `AVMediaType` was introduced in 10.13, and is [documented as an alias](https://developer.apple.com/documentation/avfoundation/avmediatype?language=objc), so probably for a while unless there’s explicit notice that depending on it being an alias is deprecated?